tron/core/action.py

```diff
@@ -107,30 +52,29 @@ def from_config(cls, config):
             executor=config.executor,
             cpus=config.cpus,
             mem=config.mem,
-            constraints=constraints,
             docker_image=config.docker_image,
-            docker_parameters=docker_parameters,
-            env=config.env,
-            extra_volumes=extra_volumes,
             trigger_downstreams=config.trigger_downstreams,
             triggered_by=config.triggered_by,
             on_upstream_rerun=config.on_upstream_rerun,
         )
 
-    def __eq__(self, other):
-        attributes_match = all(
-            getattr(self, attr, None) == getattr(other, attr, None)
-            for attr in self.equality_attributes
-        )
-        return attributes_match and all(
-            self_act == other_act for (
-                self_act,
-                other_act,
-            ) in zip(self.required_actions, other.required_actions)
-        )
+        # Only convert config values if they are not None.
+        constraints = config.constraints
+        if constraints:
+            constraints = set((c.attribute, c.operator, c.value) for c in constraints)
```

Comment:
Do we need to turn them into normal tuples, if the constraint/parameter/volume configs are already namedtuples?